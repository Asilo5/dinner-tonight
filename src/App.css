* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}


.App {
  text-align: center;
}

.App-header {
  color: #ff3838;
  font-size: 5em;
  font-family: 'Nothing You Could Do', cursive;
}

.App-link {
  color: #09d3ac;
}
