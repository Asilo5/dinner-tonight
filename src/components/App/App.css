* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

body {
  background-color: #FDDFD5;
}

.App {
  text-align: center;
}

.App-header {
  color: #ff3838;
  font-size: 5em;
  font-family: 'Nothing You Could Do', cursive;
}

.App-link {
  color: #09d3ac;
}
