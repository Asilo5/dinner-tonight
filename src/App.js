import React, { Component } from 'react';
import './App.css';
import FormSearch from './FormSearch/FormSearch';
// import RecipeContainer from './RecipeContainer/RecipeContainer';

const API_KEY = '2328b6e1ad7d799035a86bee44248fd1';

class App extends Component {
  constructor() {
    super();
    this.state = {
      recipes: []
    }
  }


  getRecipe = async (e) => {
    e.preventDefault(); 
    const recipeName = e.target.elements.recipeName.value;
    console.log(recipeName);
    
    const api_call = await fetch(`https://www.food2fork.com/api/search?key=${API_KEY}&q=${recipeName}&count=10`);
    
    const data = await api_call.json(); 
    this.setState({ recipes : data.recipes });
    console.log(this.state.recipes);
  }


  render() {
    return (
      <section className='App'>
        <header className='App-header'>
          <h1> Recipe App </h1>
        </header>
        <FormSearch getRecipe={this.getRecipe}/>
        {this.state.recipes.map((recipe) => {
          return <p key={recipe.recipe_id}>{recipe.title}</p>
        })}
        {/* <RecipeContainer recipes={this.state.recipes}/> */}
      </section>
    )
  }
}

export default App;
