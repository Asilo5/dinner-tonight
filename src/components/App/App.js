import React, { Component } from 'react';
import './App.css';
import FormSearch from '../FormSearch/FormSearch';
import RecipeContainer from '../RecipeContainer/RecipeContainer';

const API_KEY = '2328b6e1ad7d799035a86bee44248fd1';

class App extends Component {
  constructor() { 
    super();
    this.state = {
      recipes: []
    }
  }

  getRecipe = async (ingredient) => {
    const api_call = await fetch(`https://www.food2fork.com/api/search?key=${API_KEY}&q=${ingredient}&count=12`);
    const data = await api_call.json(); 
    this.setState({ recipes : data.recipes });
  }

  render() {
    return (
      <section className='App'>
        <header className='App-header'>
          <h1> Dinner Tonight </h1>
        </header>
        <FormSearch getRecipe={this.getRecipe}/>
        { this.state.recipes.length === 0 ? <p className='empty-message'> Please search for your dinner first.. </p> : <RecipeContainer recipes={this.state.recipes}/>}
      </section>
    )
  }
}

export default App;
