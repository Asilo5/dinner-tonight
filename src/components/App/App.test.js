import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {

  let wrapper;

  let mockRecipe = [
    {
      image_url: "http://static.food2fork.com/BananaPeanutButterChocolateChipCookies13e87.jpg",
      publisher: "Two Peas and Their Pod",
      publisher_url: "http://www.twopeasandtheirpod.com",
      recipe_id: "ed9409",
      source_url: "http://www.twopeasandtheirpod.com/gluten-free-vegan-banana-peanut-butter-chocolate-chip-cookies/",
      title: "Gluten-Free Vegan Banana Peanut Butter Chocolate Chip Cookies",
    }
  ]

  beforeEach(() => {
    wrapper = shallow(<App />);
  })

  it('should match snapshot with all information getting passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  // come back to this test when we go over testing async await

  it.skip('should update state when getRecipe is called', () => {
    expect(wrapper.state('recipes')).toEqual([])
    wrapper.instance().getRecipe('vegan');
    expect(wrapper.state('recipes')).toEqual(mockRecipe);
  })

})
