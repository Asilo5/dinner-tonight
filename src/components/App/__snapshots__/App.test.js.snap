// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot with all information getting passed in correctly 1`] = `
<section
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
       Dinner Tonight 
    </h1>
  </header>
  <FormSearch
    getRecipe={[Function]}
  />
  <p
    className="empty-message"
  >
     Please search for your dinner first.. 
  </p>
</section>
`;
